docs/source/tutorial/tutorial-execute-query.md

```diff
@@ -0,0 +1,124 @@
+
+## Running code generation
+
+1. Return to your project's **Apollo CLI** Run Script build phase. Comment out the line that you added to the bottom (that includes `apollo:schema`) and _uncomment_ the line you previously commented out (that includes `codegen:generate`).
+
+2. Build your project. When the build completes, an `API.swift` file appears in the same folder  as `schema.json`. 
+
+3. Drag the `API.swift` file into Xcode. This time, **do** check the **Add to target** box for the `RocketReserver` app. You include this file in your application's bundle to enable you to execute the query you defined. 
+
+### The `API.swift` file
+
+Open the `API.swift` file. It defines a root class, `LaunchListQuery`, with many nested structs below it. If you compare the structs to the JSON data returned in GraphiQL, you see that the structure matches. These structs include properties only for the fields that your query requests. 
+
+Try commenting out the `id` property in `LaunchList.graphql`, saving, then building again. When the build completes, the innermost `Launch` now only includes the built-in `__typename` and the requested `site` property. 
+
+Uncomment `id` and rebuild to restore the property.
+
+Now that you've generated code and had a chance to see what's in there, it's time to get everything working end to end!
+
+## Running a test query
+
+To use the generated operations in `API.swift`, you first create an instance of `ApolloClient`. This instance takes your generated code and uses it to make network calls to your server. It's recommended that this instance is a singleton or static instance that's accessible from anywhere in your codebase.
+
+1. Create a new Swift file called `Network.swift` and copy the code from [Basic client creation](/initialization/#basic-client-creation) into it. Make sure to add `import Apollo` to the top of the file.
+
+2. Update the URL string to be `https://n1kqy.sse.codesandbox.io/` instead of the `localhost` URL in the example.
+
+3. To make sure your `ApolloClient` instance is communicating correctly with the server, add the following code to `AppDelegate.swift` in the `application:didFinishLaunchingWithOptions` method, above `return true`:
+
+    ```swift
+    Network.shared.apollo.fetch(query: LaunchListQuery()) { result in
+      switch result {
+      case .success(let graphQLResult):
+        print("Success! Result: \(graphQLResult)")
+      case .failure(let error):
+        print("Failure! Error: \(error)")
+      }
+    }
+```
+
+Build and run your application. CodeSandbox might take a few seconds to spin up your GraphQL server if nobody's been using it recently, but once it's up, you should see a response that resembles the following: 
+
+![success log output](images/success_log_barf.png)
+
+You're now successfully fetching data from the network using generated code! Now it's time to move on to [displaying query results in your UI](tutorial_2).
```

Comment:
since you took out the links at the bottom of the other tutorials, should we get rid of this one as well?

Reply:
Ah, hm, I'll add links back to the others instead :D